src/Npgsql/TypeMapping/TypeMapperBase.cs

```diff
@@ -33,7 +35,7 @@ abstract class TypeMapperBase : INpgsqlTypeMapper
     {
         internal Dictionary<string, NpgsqlTypeMapping> Mappings { get; set; }
 
-        public INpgsqlNameTranslator DefaultNameTranslator { get; set; }
+        public abstract INpgsqlNameTranslator DefaultNameTranslator { get; }
```

Comment:
I may be missing something, but why not keep this a regular, non-virtual property on `TypeMapperBase` and make the setter protected? This would save having have two overrides of this property in `GlobalTypeMapper` and `ConnectorTypeMapper`...

Reply:
Oh, that's a silly oversight. Though, instead of a protected setter, it seems safer to pass this down to a base constructor.